staging/src/k8s.io/kubectl/pkg/cmd/create/create_namespace.go

```diff
@@ -36,16 +49,42 @@ var (
 	  kubectl create namespace my-namespace`))
 )
 
-// NamespaceOpts is the options for 'create namespace' sub command
-type NamespaceOpts struct {
-	CreateSubcommandOptions *CreateSubcommandOptions
+// NamespaceOptions is the options for 'create namespace' sub command
+type NamespaceOptions struct {
+	// PrintFlags holds options necessary for obtaining a printer
+	PrintFlags *genericclioptions.PrintFlags
+	// Name of resource being created
+	Name string
+	// StructuredGenerator is the resource generator for the object being created
+	StructuredGenerator generate.StructuredGenerator
+	DryRunStrategy      cmdutil.DryRunStrategy
+	DryRunVerifier      *resource.DryRunVerifier
+	CreateAnnotation    bool
+	FieldManager        string
+
+	Namespace        string
+	EnforceNamespace bool
+
+	Mapper        meta.RESTMapper
+	DynamicClient dynamic.Interface
```

Comment:
I guess we can probably replace the Dynamicclient here with the normal client: https://github.com/kubernetes/client-go/tree/master/kubernetes/typed/core/v1